cairo/src/utils/bytes.cairo

```diff
@@ -288,77 +288,132 @@ func bytes_to_bytes8_little_endian{range_check_ptr}(dst: felt*, bytes_len: felt,
         return ();
     }
 
-    let (local pow256) = get_label_location(pow256_table);
-    let (full_u64_word_count, local last_input_num_bytes) = unsigned_div_rem(bytes_len, 8);
-    local range_check_ptr = range_check_ptr;
-
-    tempvar dst_index = 0;
-    tempvar bytes_index = bytes_len - 1;
-    tempvar bytes8 = 0;
-    tempvar bytes8_index = 7;
-
-    body:
-    let dst_index = [ap - 4];
-    let bytes_index = [ap - 3];
-    let bytes8 = [ap - 2];
-    let bytes8_index = [ap - 1];
-
-    let bytes_len = [fp - 4];
-    let bytes = cast([fp - 3], felt*);
-    let pow256 = cast([fp], felt*);
-    let current_byte = bytes[bytes_len - 1 - bytes_index];
-    let current_pow = pow256[bytes8_index];
+    tempvar less_than_8;
+    %{ bytes_len_less_than_8 %}
+    tempvar bytes8 = dst;
+    tempvar bytes = bytes;
 
-    tempvar bytes8 = bytes8 + current_byte * current_pow;
+    static_assert bytes8 == [ap - 2];
+    static_assert bytes == [ap - 1];
+
+    jmp skip_full_word_loop if less_than_8 != 0;
+
+    full_word_loop:
+    let bytes8 = cast([ap - 2], felt*);
+    let bytes = cast([ap - 1], felt*);
+
+    assert [bytes8] = bytes[0] + bytes[1] * 256 + bytes[2] * 256 ** 2 + bytes[3] * 256 ** 3 + bytes[
```

Comment:
i think we could benefit from adding soundness assumptions as comments in the codebase. here we know that if the prover lied on less_than_8 then accessing bytes[i], i <= will cause a memory error

Reply:
or at least have this as function doc